public/app/plugins/datasource/elasticsearch/dataquery.gen.ts

```diff
@@ -0,0 +1,368 @@
+
+export type BucketAggregationType = ('terms' | 'filters' | 'geohash_grid' | 'date_histogram' | 'histogram');
+
+export interface BaseBucketAggregation {
+  id: string;
+  settings?: Record<string, unknown>;
+  type: BucketAggregationType;
+}
+
+export interface BucketAggregationWithField extends BaseBucketAggregation {
+  field?: string;
+}
+
+export interface DateHistogram extends BucketAggregationWithField {
+  type: 'date_histogram';
+}
+
+export interface DateHistogramSettings {
+  interval?: string;
+  min_doc_count?: string;
+  offset?: string;
+  timeZone?: string;
+  trimEdges?: string;
+}
+
+export interface Histogram extends BucketAggregationWithField {
+  settings?: {
+    interval: string;
+    min_doc_count?: string;
+  };
+  type: 'histogram';
+}
+
+export interface HistogramSettings {
+  interval: string;
+  min_doc_count?: string;
+}
+
+export type TermsOrder = ('desc' | 'asc');
+
+export interface Terms extends BucketAggregationWithField {
+  type: 'terms';
+}
+
+export interface TermsSettings {
+  min_doc_count?: string;
+  missing?: string;
+  order?: TermsOrder;
+  orderBy?: string;
+  size?: string;
+}
+
+export interface Filters extends BaseBucketAggregation {
+  settings?: {
+    filters: Array<Filter>;
+  };
+  type: 'filters';
+}
+
+export type Filter = {
+  query: string,
+  label: string,
+};
+
+export interface FiltersSettings {
+  filters: Array<Filter>;
+}
+
+export const defaultFiltersSettings: Partial<FiltersSettings> = {
+  filters: [],
+};
+
+export interface GeoHashGrid extends BucketAggregationWithField {
+  type: 'geohash_grid';
+}
+
+export interface GeoHashGridSettings {
+  precision?: string;
+}
+
+export type PipelineMetricAggregationType = ('moving_avg' | 'moving_fn' | 'derivative' | 'serial_diff' | 'cumulative_sum' | 'bucket_script');
+
+export type MetricAggregationType = ('count' | 'avg' | 'sum' | 'min' | 'max' | 'extended_stats' | 'percentiles' | 'cardinality' | 'raw_document' | 'raw_data' | 'logs' | 'rate' | 'top_metrics' | PipelineMetricAggregationType);
+
+export interface BaseMetricAggregation {
+  hide?: boolean;
+  id: string;
+  type: MetricAggregationType;
+}
+
+export interface PipelineVariable {
+  name: string;
+  pipelineAgg: string;
+}
+
+export interface MetricAggregationWithField extends BaseMetricAggregation {
+  field?: string;
+}
+
+export interface MetricAggregationWithMissingSupport extends BaseMetricAggregation {
+  settings?: {
+    missing?: string;
+  };
+}
+
+export type InlineScript = (string | {
+    inline?: string
+  });
+
+export interface MetricAggregationWithInlineScript extends BaseMetricAggregation {
+  settings?: {
+    script?: InlineScript;
+  };
+}
+
+export interface Count extends BaseMetricAggregation {
+  type: 'count';
+}
+
+export interface Average extends MetricAggregationWithField, MetricAggregationWithMissingSupport, MetricAggregationWithInlineScript {
+  field?: string;
+  settings?: {
+    script?: (InlineScript | InlineScript);
```

Comment:
@sdboyer seems like there's smth weird going on with generated types, this should be `script?: InlineScript`, it's defined in cue as `script?:  #InlineScript`. Not sure where the union comes from.

Reply:
yeah this is...really weird. i'll have to dig into it

fortunately i think - unless it's actually invalid according to tsc - it's not harmful